### PR TITLE
[release-v0.16] chore: Bump dependencies

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

Bump following dependencies to latest supported versions.

- sigs.k8s.io/controller-runtime
- k8s.io/*
- github.com/emicklei/go-restful

Afterwards run make generate and make manifests.

**Which issue(s) this PR fixes**: 

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -4,16 +4,15 @@ go 1.18
 
 require (
 	github.com/openshift/api v0.0.0-20220124143425-d74727069f6f // release-4.10
-	k8s.io/apimachinery v0.23.5
+	k8s.io/apimachinery v0.25.15
```
Comment (akrejcir) [inline api/go.mod:7]: Is this minor version update needed? Isn't there a `v0.23.x` version with the fix?

File: api/go.mod [hunk 2]
```diff
@@ -4,16 +4,15 @@ go 1.18
...
 	kubevirt.io/containerized-data-importer-api v1.53.1
 	kubevirt.io/controller-lifecycle-operator-sdk/api v0.2.4
-	sigs.k8s.io/controller-runtime v0.11.2
+	sigs.k8s.io/controller-runtime v0.13.2
```
Comment (akrejcir) [inline api/go.mod:10]: The same question here.

File: api/go.mod (hunk 1 above)
Comment (0xFelix) [reply api/go.mod:7]: The oldest release branch with the fix is `v0.25.15`.

File: api/go.mod (hunk 2 above)
Comment (0xFelix) [reply api/go.mod:10]: Same, oldest release branch with the fix is `v0.13.2`.

**Pull request merged**